src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,81 @@
-export const Pagination = () => {};
+import React, { Dispatch, SetStateAction } from 'react';
+
+type Props = {
+  currentPage: number;
+  total: number;
+  itemsPerPage: number;
+  onPageChange: (page: number) => void;
+  setCurrentPage: Dispatch<SetStateAction<number>>;
+};
+
+export const Pagination: React.FC<Props> = ({
+  currentPage,
+  total,
+  itemsPerPage,
+  onPageChange,
+  setCurrentPage,
+}) => {
+  const pages: number[] = [];
+
+  for (let i = 1; i <= Math.ceil(total / itemsPerPage); i++) {
+    pages.push(i);
+  }
+
+  const onNextPage = () => {
+    if (currentPage !== pages.length) {
+      setCurrentPage(currentPage + 1);
```

Comment:
Instead of using `setCurrentPage`, you should call `onPageChange(currentPage + 1)` to handle page changes.